data_diff/diff_tables.py

```diff
@@ -253,6 +280,28 @@ def diff_tables(self, table1: TableSegment, table2: TableSegment) -> DiffResult:
 
         return self._bisect_and_diff_tables(table1, table2)
 
+    def _validate_and_adjust_columns(self, table1, table2):
```

Comment:
A few tests for some of this would be great in `test_diff_tables.py`

Reply:
`test_types` already tests it. Only the missing columns aren't tested, but we don't have a test for every exception in our code. (far from it)